sklearn/ensemble/iforest.py

```diff
@@ -177,12 +216,15 @@ def fit(self, X, y=None, sample_weight=None):
             max_samples = int(self.max_samples * X.shape[0])
 
         self.max_samples_ = max_samples
-        max_depth = int(np.ceil(np.log2(max(max_samples, 2))))
+
+        self._validate_estimator()
+
         super(IsolationForest, self)._fit(X, y, max_samples,
-                                          max_depth=max_depth,
+                                          max_depth=self.max_depth_,
```

Comment:
`BaseEnsemble` has a list of parameters to copy via `estimator_params` which is used in `forest.py`. Could this be another way to make the code less special?